src/openzeppelin/governance/timelock/library.cairo

```diff
@@ -0,0 +1,505 @@
+
+        let (pending: felt) = is_operation_pending(id)
+        let (ready: felt) = is_le(timestamp, block_timestamp)
+
+        if pending + ready == 2:
+            return (TRUE)
+        else:
+            return (FALSE)
+        end
+    end
+
+    func is_operation_done{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(id: felt) -> (done: felt):
+        let (timestamp: felt) = get_timestamp(id)
+
+        if timestamp == DONE_TIMESTAMP:
+            return (TRUE)
+        else:
+            return (FALSE)
+        end
+    end
+
+    func update_delay{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(min_delay: felt):
+        let (old_min_delay: felt) = Timelock_min_delay.read()
+        Timelock_min_delay.write(min_delay)
+        MinDelayChange.emit(old_min_delay, min_delay)
+
+        return ()
+    end
+
+    func schedule{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(
+        call_array_len: felt,
+        call_array: TimelockCall*,
+        calldata_len: felt,
+        calldata: felt*,
+        predecessor: felt,
+        salt: felt,
+        delay: felt,
+    ):
+        alloc_locals
+
+        let (id: felt) = hash_operation(call_array_len, call_array, calldata, predecessor, salt)
+
+        let (calls: Call*) = alloc()
+        _from_timelock_calls_to_calls(call_array_len, call_array, calldata, calls)
+        _schedule(id, delay, call_array_len, calls, predecessor)
+
+        return ()
+    end
+
+    func execute{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(
+        call_array_len: felt,
+        call_array: TimelockCall*,
+        calldata_len: felt,
+        calldata: felt*,
+        predecessor: felt,
+        salt: felt,
+    ):
+        alloc_locals
+
+        let (id: felt) = hash_operation(call_array_len, call_array, calldata, predecessor, salt)
+        let (ready: felt) = is_operation_ready(id)
+
+        with_attr error_message("Timelock: operation is not ready"):
+            assert ready = TRUE
+        end
+
+        if predecessor != 0:
+            let (predecessor_done: felt) = is_operation_done(predecessor)
+
+            with_attr error_message("Timelock: missing dependency"):
+                assert predecessor_done = TRUE
+            end
+        end
+
+        _execute(id, call_array_len, call_array, calldata_len, calldata, predecessor)
+
+        Timelock_timestamps.write(id, 1)
+
+        return ()
+    end
+
+    func cancel{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(id: felt):
+        _cancel(id)
+
+        Cancelled.emit(id)
+
+        return ()
+    end
+
+    func _schedule{
+        syscall_ptr: felt*, 
+        pedersen_ptr: HashBuiltin*, 
+        range_check_ptr
+    }(
```

Comment:
do you think it makes sense to have a separate `_schedule` function? I think in Solidity that's the case only to implement `schedule` and `scheduleBatch` but that's not the case here.